internal/services/v1/relationships.go

```diff
@@ -162,31 +161,23 @@ func (ps *permissionServer) WriteRelationships(ctx context.Context, req *v1.Writ
 
 	// Check for duplicate updates and create the set of caveat names to load.
 	updateRelationshipSet := util.NewSet[string]()
-	referencedCaveatNamesWithContext := util.NewSet[string]()
 	for _, update := range req.Updates {
 		tupleStr := tuple.StringRelationship(update.Relationship)
 		if !updateRelationshipSet.Add(tupleStr) {
 			return nil, rewriteError(
 				ctx,
-				status.Errorf(
-					codes.InvalidArgument,
-					"found duplicate update operation for relationship %s",
-					tupleStr,
-				),
+				NewDuplicateRelationshipErr(update),
 			)
 		}
 
-		// Only load the caveat if we need its type information for context checking.
-		if hasNonEmptyCaveatContext(update) {
-			if !ps.caveatsEnabled {
+		if !ps.caveatsEnabled {
+			if update.Relationship.OptionalCaveat != nil && update.Relationship.OptionalCaveat.CaveatName != "" {
 				return nil, fmt.Errorf("caveats are currently not supported")
```

Comment:
shouldn't we create a new typed error here and rewrite it? This will end up in gRPC code `UNKNOWN`.

Reply:
Which is fine for now; this is temporary until such time until the flag is removed